server.js

```diff
@@ -3,35 +3,85 @@ import bodyParser from 'body-parser'
 import cors from 'cors'
 import mongoose from 'mongoose'
 
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// 
-// import goldenGlobesData from './data/golden-globes.json'
-// import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
-// import topMusicData from './data/top-music.json'
-
-const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo"
+import goldenGlobesData from './data/golden-globes.json'
+//console.log(goldenGlobesData)
+const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo-anna"
 mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
 mongoose.Promise = Promise
 
+// Properties: key & value-pair. Constructor = new mongoose.model(). Function that
+// takes two arguments: a string and an object. Calling a function with two arg.
+// The object is called schema. Then we populate with instances of model with the
+// properties of object.
+const Nomination = new mongoose.model('Nomination', {
+  year_film: Number,
+  year_award: Number,
+  ceremony: Number,
+  category: String,
+  nominee: String,
+  film: String,
+  win: Boolean
+});
+
+// Start MongoDB: RESET_DB=true npm run dev
+// To populate database: remove if-statement, push, put if-statement back, push again
+if (process.env.RESET_DB) {
+	const seedDatabase = async () => {
+    await Nomination.deleteMany();
+		goldenGlobesData.forEach(item => {
+      const newNomination = new Nomination(item);
+      newNomination.save();
+		})
+  }
+  seedDatabase();
+}
+
 // Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
+// overridden when starting the server. For example: PORT=9000 npm start
 const port = process.env.PORT || 8080
 const app = express()
 
 // Add middlewares to enable cors and json body parsing
 app.use(cors())
 app.use(bodyParser.json())
 
+// express starts by using the middleware, check if database is connected. If ok,
+// express goes on to first route. If not, it returns error 503
+
+  app.use((req, res, next) => {
+  if (mongoose.connection.readyState === 1) {
+    next()
+  } else {
+    res.status(503).json({ error: 'Service unavailable '})
+  }
+})
+
 // Start defining your routes here
 app.get('/', (req, res) => {
   res.send('Hello world')
 })
 
+app.get('/nominations', async (req, res) => {
+    //const queryParameters = req.query;
+    const allNominations = await Nomination.find(req.query);
+    if (allNominations) {
+      res.json(allNominations)
+    } else {
+      res.status(404).json({ error: 'Nominations not found'})
+    }
+})
+
+app.get('/nominations/:nominee', async (req, res) => {
```

Comment:
`nominee` is not a good key for this endpoint, unfortunately. When you use it like this you returning a bad set of data because you cutting off a big portion of it. For example "Avatar" has a lot of nominations, but you are showing only the first one. 

You need to use field which is always unique for every document, usually, it is `id` field.

Reply:
I understand but how do I do it when the data doesn't contain an id field?

Reply:
If you want to use /:nominee you can do it, but you shouldn't do it together with a method that only returns a single result (the first match, even though there are many matches). As it is a requirement for this project to have an endpoint that returns a single result, you could keep this endpoint and change it to display all nominations with the help of another method but you also have to create an endpoint that returns a single result where it actually _is_ only one result e.g. a specific category, a specific year where win=true.